### ISSUE TITLE
in expected, regions are expected to be integer-indexed

### ISSUE THREAD

**Issue by golobor** [open2c/cooltools#184]
https://github.com/open2c/cooltools/blob/74a767eb6a101928a33d8dbcca2fd3c489092a87/cooltools/expected.py#L626

**Comment by sergpolly** [comment]
by @gfudenberg  from slack:

> found a funny behavior of diagsum-- it seems to assume that the regions df has index starting with index 0, and throws a strange error if you pass e.g. regions[1:] without creating a new regions df and re-setting index (https://github.com/open2c/cooltools/blob/43448008d1a1b4b7ebff8a768427918a0100c18c/cooltools/expected.py#L626) not sure on best fix/change in documentation given ongoing refactoring

**Comment by golobor** [comment]
ok, we figured that there was a discrepancy: assign_supports returns an integer-based index of the support region, which is then used against the (potentially, non-integer) index of view_df. This can result in error, when view_df is sliced. 
Proposed solution:
replace
https://github.com/open2c/cooltools/blob/704f259c4df5bc1c61be915bf56c80b00a0298ed/cooltools/expected.py#L677
with

`view_df["name"].iat[i]`

**Comment by sergpolly** [comment]
unrelated, for myself: check in the new expected code if we're making any `pixels.dropna()`  - as those aren't fool proof against annotated pixels with several balancing weight columns, or when working with raw data

**Comment by sergpolly** [comment]
`pixels.dropna()` is now `pixels.dropna(subset=["r1","r2"])` - i.e. fool-proof - resolved

**Comment by sergpolly** [comment]
#296 would resolve it

**Comment by sergpolly** [comment]
after consideration had to do 
```python
    if weight_name is None:
        pixels = pixels.dropna(subset=["r1", "r2"])
    else:
        pixels = pixels.dropna(subset=["r1", "r2", weight_name])
```
in order to calculate raw "conut" sum only over valid pixels !

old `dropna()` was taking care of both filtering pixels to unassigned supports and anything with weight=NaN
if just wasn't obvious at all

**Comment by sergpolly** [comment]
resolved in #296
